lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1597,6 +1597,15 @@ public function addToIdentityMap($entity)
         $className     = $classMetadata->rootEntityName;
 
         if (isset($this->identityMap[$className][$idHash])) {
+            if ($this->identityMap[$className][$idHash] !== $entity) {
+                throw new RuntimeException(sprintf(
+                    'While adding an entity of class %s with an ID hash of "%s" to the identity map, another object of class %s was already present for the same ID. This exception is a safeguard against an internal inconsistency - IDs should uniquely map to entity object instances. This problem may occur if you use application-provided IDs and reuse ID values. Or, if database-provided IDs are reassigned after truncating the database without clearing the EntityManager. Or, you might have been using EntityManager#getReference() to create a reference for a nonexistent ID that was subsequently (by the RDBMS) assigned to another entity. Otherwise, it might be an ORM-internal inconsistency, please report it.',
```

Comment:
Did you consider using linebreaks here? If yes, what made you avoid them?

Reply:
Yes – that I did not find a good example right away that would show me what the desired code style is in that case.

Reply:
It might be personal but I'd go with wrapping at the 80th column, and use nowdoc so as not to have to worry about quotes.

Reply:
Example: https://github.com/doctrine/orm/blob/55d477dc507a52675c5a61267a00d389c71182f4/lib/Doctrine/ORM/Exception/NotSupported.php#L19-L23

Reply:
I don't like nowdocs because they break out of indentation, suddenly all text is at the left

Reply:
What about so?

```diff
diff --git a/lib/Doctrine/ORM/UnitOfWork.php b/lib/Doctrine/ORM/UnitOfWork.php
index 4b15fa47a0..f31eeaffec 100644
--- a/lib/Doctrine/ORM/UnitOfWork.php
+++ b/lib/Doctrine/ORM/UnitOfWork.php
@@ -1599,7 +1599,18 @@ class UnitOfWork implements PropertyChangedListener
         if (isset($this->identityMap[$className][$idHash])) {
             if ($this->identityMap[$className][$idHash] !== $entity) {
                 throw new RuntimeException(sprintf(
-                    'While adding an entity of class %s with an ID hash of "%s" to the identity map, another object of class %s was already present for the same ID. This exception is a safeguard against an internal inconsistency - IDs should uniquely map to entity object instances. This problem may occur if you use application-provided IDs and reuse ID values. Or, if database-provided IDs are reassigned after truncating the database without clearing the EntityManager. Or, you might have been using EntityManager#getReference() to create a reference for a nonexistent ID that was subsequently (by the RDBMS) assigned to another entity. Otherwise, it might be an ORM-internal inconsistency, please report it.',
+                    'While adding an entity of class %s with an ID hash of "%s" ' .
+                    'to the identity map, another object of class %s was already ' .
+                    'present for the same ID. This exception is a safeguard against ' .
+                    'an internal inconsistency - IDs should uniquely map to entity ' .
+                    'object instances. This problem may occur if you use ' .
+                    'application-provided IDs and reuse ID values. Or, if database-provided ' .
+                    'IDs are reassigned after truncating the database without ' .
+                    'clearing the EntityManager. Or, you might have been using ' .
+                    'EntityManager#getReference() to create a reference for a ' .
+                    'nonexistent ID that was subsequently (by the RDBMS) assigned ' .
+                    'to another entity. Otherwise, it might be an ORM-internal ' .
+                    'inconsistency, please report it.',
                     get_class($entity),
                     $idHash,
                     get_class($this->identityMap[$className][$idHash])
```

Reply:
> I don't like nowdocs because they break out of indentation, suddenly all text is at the left

Yeah, but that's not the case with recent versions of PHP, right? So I think it would be better to use nowdoc early, because on 3.0.x you will just have to indent everything instead of resorting to concatenation.

> What about so?

I see 2 issues with this:

- it's harder to maintain than nowdoc
- the output text has no linebreaks.

People reading the error message would want linebreaks as well in this case, I think. In recent versions of PHP, with nowdoc, you can have such strings with linebreaks _without introducing extra indentation in output text_.

Reply:
TIL!

I've changed it to a nowdoc.

Reply:
It's beautiful!